src/JsonService.ts

```diff
@@ -27,6 +27,30 @@ export class JsonService {
         }
     }
 
+    protected async fetchWithTimeout(input: RequestInfo, init: RequestInit & { timeoutInSeconds?: number } = {}) {
+        const { timeoutInSeconds, ...initFetch } = init;
+        if (!timeoutInSeconds) {
+            return await fetch(input, initFetch);
+        }
+
+        try {
+            const controller = new AbortController();
+            const timeoutId = setTimeout(() => controller.abort(), timeoutInSeconds * 1000);
+            const response = await fetch(input, {
+                ...init,
+                signal: controller.signal,
+            });
+            clearTimeout(timeoutId);
```

Comment:
I think this line would be better in a `finally` block. Lines `37` and `38` can be moved outside of the `try` since they aren't expected to throw.